app/scripts/metamask-controller.js

```diff
@@ -6597,21 +6597,7 @@ export default class MetamaskController extends EventEmitter {
         this.networkController,
       ),
       setActiveNetwork: async (networkClientId) => {
-        // if the origin has the CAIP-25 permission
-        // we set per dapp network selection state
-        if (
-          this.permissionController.hasPermission(
```

Comment:
we have to be carefull here , removing this code can introduce a bug on the dapp connectivity , checking now